src/main/java/app/AppBuilder.java

```diff
@@ -267,9 +273,19 @@ public AppBuilder addSignupUseCase() {
      */
     public AppBuilder addListUseCase() {
         listPresenter = new ListPresenter(listViewModel);
+
         this.listInteractor = new ListInteractor(userDataAccessObject, listPresenter);
+        this.removeInteractor = new RemoveInteractor(userDataAccessObject, listPresenter);
+        this.moveInteractor = new MoveInteractor(userDataAccessObject, listPresenter);
+
         final ListController listController = new ListController(listInteractor);
+        final RemoveController removeController = new RemoveController(removeInteractor);
+        final MoveController moveController = new MoveController(moveInteractor);
+
         listView.setListController(listController);
+        listView.setRemoveController(removeController);
+        listView.setMoveController(moveController);
+
```

Comment:
Hey Judy! Nice progress on refactoring and adding the move-media use case, I think it looks good and fits with CA! I just wanted to suggest that you also move the construction of the remove-media and move-media use cases into their own methods within the appbuilder (so as not to violate Single-responsibility principle). My suggestion below is just a suggestion, feel free to do things your way. Also, I went through the usages for the interactors, but outside of this method in the AppBuilder there's no other usages. So, instead of having them be instance variables of the Appbuilder, I think it would be fine to set them as inline variables instead (to reduce their scope). 

```suggestion

        final ListInteractor listInteractor = new ListInteractor(userDataAccessObject, listPresenter);
        final ListController listController = new ListController(listInteractor);
        listView.setListController(listController);
        
        return this;
    }    
   
    /**
     * Adds the Move Media Use Case to the application, allowing users to move a piece of media from one collection
     * to another.
     * @return this builder
     */
    public AppBuilder addMoveMediaUseCase() {
        if (listView == null) {
            throw new RuntimeException("addListView must be called before addMoveMediaUseCase");
        }
        if (listPresenter == null) {
            throw new RuntimeException("addListUsecase must be called before addMoveMediaUseCase");
        }
     
        final MoveInteractor moveInteractor = new MoveInteractor(userDataAccessObject, listPresenter);
        final MoveController moveController = new MoveController(moveInteractor);
        listView.setMoveController(moveController);
        
        return this;
    }
    
    /**
     * Adds the Remove Media Use Case to the application, allowing users to remove a piece of media from the 
     * selected collection.
     * @return this builder
     */
    public AppBuilder addRemoveMediaUseCase() {
        if (listView == null) {
            throw new RuntimeException("addListView must be called before addRemoveMediaUseCase");
        }
        if (listPresenter == null) {
            throw new RuntimeException("addListUsecase must be called before addRemoveMediaUseCase");
        }
     
        final RemoveInteractor removeInteractor = new RemoveInteractor(userDataAccessObject, listPresenter);
        final RemoveController removeController = new RemoveController(removeInteractor);
        listView.setRemoveController(removeController);

```